### PR TITLE
gpu: Fix static array checking

### PR DESCRIPTION
**Pull request by spencer-lunarg**
closes https://github.com/KhronosGroup/Vulkan-ValidationLayers/issues/6944

Now that GPU-AV is in a better shape, was able to actually try and re-work this logic. Tests added and re-enabled to confirm things work as expected (still some open edge cases, but will be worked out after getting some spec clarification)

### CODE REVIEW

File: layers/gpu/spirv/descriptor_indexing_oob_pass.cpp [hunk 1]
```diff
@@ -109,68 +108,99 @@ void DescriptorIndexingOOBPass::Reset() {
...
+        }
+        var_inst_ = &variable->inst_;
+
+        const Type* pointer_type = variable->PointerType(module_.type_manager_);
+        if (!pointer_type) {
+            module_.InternalError(Name(), "Pointer type not found");
+            return false;
+        }
+
+        if (pointer_type->inst_.IsArray() && access_chain_inst->Length() >= 5) {
```
Comment (arno-lunarg) [inline layers/gpu/spirv/descriptor_indexing_oob_pass.cpp:139]: Some comment on this 5 maybe?

File: layers/gpu/spirv/descriptor_indexing_oob_pass.cpp [hunk 2]
```diff
@@ -109,68 +108,99 @@ void DescriptorIndexingOOBPass::Reset() {
...
         }
 
-        const Variable* variable = module_.type_manager_.FindVariableById(access_chain_inst_->Operand(0));
+        const Variable* variable = module_.type_manager_.FindVariableById(access_chain_inst->Operand(0));
         if (!variable) {
             return false;
         }
         var_inst_ = &variable->inst_;
 
         uint32_t storage_class = variable->StorageClass();
```
Comment (arno-lunarg) [inline layers/gpu/spirv/descriptor_indexing_oob_pass.cpp:161]:
```suggestion
        const uint32_t storage_class = variable->StorageClass();
```

**Review by arno-lunarg — APPROVED** [review]
